File: A02/Python/P02/d.-RaizCuadrada.py
import time
import math

def format_time(t):
    time_formatted = ""
    hour = minute = second = milisecond = int(0)

    hour = int(t//3600);
    t = t - (hour*3600)
    minute = int(t//60)
    t = t - (minute*60)
    second = int(t//1)
    t = t - second
    t = int(t*10000)
    milisecond = t
    
    time_formatted += (("0" + str(hour)) if hour < 10 else str(hour)) + ':'
    time_formatted += (("0" + str(minute)) if minute < 10 else str(minute)) + ':'
    time_formatted += (("0" + str(second)) if second < 10 else str(second)) + '.'
    time_formatted += '{:0>4}'.format(milisecond)
    
    return time_formatted


def is_prime(num):
    for i in range(2, int(math.sqrt(num)+1)):
        if num % i == 0: return False

    return True


def main():
    number = 0
    primes = 0
    exec_time = ""
    t = 0.0

    t = time.time()
    while True:
        number = int(input())
        if number == -1: break
        if is_prime(number): primes += 1
    t = time.time() - t
    exec_time = format_time(t);

    print("Total:", primes)
    print("Tiempo:", exec_time)

if __name__ == "__main__":
    main()

File: A02/C++/P04/d.-RaizCuadrada.cpp
#include <iostream>
#include <iomanip>
#include <sstream>
#include <string>
#include <ctime>
#include <cmath>
using namespace std;

string formatTime(long double t) {
	ostringstream time;
	int hour, min, sec, milisec;
	
	hour = t/3600;
	t = t - (hour*3600);
	min = t/60;
	t = t - (min*60);
	sec = t;
	t = t - (sec/1);
	milisec = t*10000;

	time << setfill('0') <<
		setw(2) << hour << ':' <<
		setw(2) << min << ':' <<
		setw(2) << sec << '.' <<
		setw(4) << milisec;
	
	return time.str();
}

bool isPrime(long num, unsigned long long &operations) {
	bool res = true;
	for(long i = 2; i <= sqrt(num); i++) {
		if (num % i == 0) res = false;
		operations++;
	}
	
	return res;
}

int main() {
	long number;
	unsigned long long operations = 0;
	int primes = 0;
	clock_t t;
	string time;
	
	t = clock();
	while(1){
		cin >> number;
		if (number == -1) break;
		primes += isPrime(number, operations);
	}
	t = clock() - t;
	time = formatTime(((long double)t)/CLOCKS_PER_SEC);

	cout << "Total: " << primes << endl;
	cout << "Operaciones: " << operations << endl;
	cout << "Tiempo: " << time << endl;

	return 0;
}
